<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1222</width>
    <height>875</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Raytracing</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../icons/icons.qrc">
    <normaloff>:/icons/graph.svg</normaloff>:/icons/graph.svg</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>10</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <property name="bottomMargin">
     <number>10</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <item>
       <widget class="SceneCanvas" name="canvas" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>1200</width>
          <height>768</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>0</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1222</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_rendering_config"/>
    <addaction name="actionSave_rendering_config"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionInit"/>
    <addaction name="actionWireframe"/>
    <addaction name="actionRender"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAbout_author"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="iconSize">
    <size>
     <width>24</width>
     <height>24</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionLoad_rendering_config"/>
   <addaction name="actionSave_rendering_config"/>
   <addaction name="separator"/>
   <addaction name="actionInit"/>
   <addaction name="actionWireframe"/>
   <addaction name="actionRender"/>
   <addaction name="separator"/>
   <addaction name="actionSettings"/>
   <addaction name="separator"/>
   <addaction name="actionAbout_author"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/document.svg</normaloff>:/icons/document.svg</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>New scene</string>
   </property>
   <property name="statusTip">
    <string>New scene</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/folder.svg</normaloff>:/icons/folder.svg</iconset>
   </property>
   <property name="text">
    <string>Open scene</string>
   </property>
   <property name="iconText">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open scene file</string>
   </property>
   <property name="statusTip">
    <string>Open scene file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionAbout_author">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/question-mark.svg</normaloff>:/icons/question-mark.svg</iconset>
   </property>
   <property name="text">
    <string>About author</string>
   </property>
   <property name="statusTip">
    <string>About author</string>
   </property>
  </action>
  <action name="actionInit">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/info.svg</normaloff>:/icons/info.svg</iconset>
   </property>
   <property name="text">
    <string>Init</string>
   </property>
   <property name="toolTip">
    <string>Initialize default scene</string>
   </property>
   <property name="statusTip">
    <string>Initialize default scene</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/browser.svg</normaloff>:/icons/browser.svg</iconset>
   </property>
   <property name="text">
    <string>Save image</string>
   </property>
   <property name="toolTip">
    <string>Save image</string>
   </property>
   <property name="statusTip">
    <string>Save image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionRender">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/image.svg</normaloff>:/icons/image.svg</iconset>
   </property>
   <property name="text">
    <string>Render</string>
   </property>
   <property name="toolTip">
    <string>Render scene</string>
   </property>
   <property name="statusTip">
    <string>Render scene</string>
   </property>
  </action>
  <action name="actionLoad_rendering_config">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/arrow-thick-top.svg</normaloff>:/icons/arrow-thick-top.svg</iconset>
   </property>
   <property name="text">
    <string>Load rendering config</string>
   </property>
   <property name="statusTip">
    <string>Load rendering config</string>
   </property>
  </action>
  <action name="actionSave_rendering_config">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/arrow-thick-bottom.svg</normaloff>:/icons/arrow-thick-bottom.svg</iconset>
   </property>
   <property name="text">
    <string>Save rendering config</string>
   </property>
   <property name="statusTip">
    <string>Save rendering config</string>
   </property>
  </action>
  <action name="actionWireframe">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/grid-four-up.svg</normaloff>:/icons/grid-four-up.svg</iconset>
   </property>
   <property name="text">
    <string>Wireframe</string>
   </property>
   <property name="toolTip">
    <string>Enter wireframe mode</string>
   </property>
   <property name="statusTip">
    <string>Enter wireframe mode</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset resource="../icons/icons.qrc">
     <normaloff>:/icons/cog.svg</normaloff>:/icons/cog.svg</iconset>
   </property>
   <property name="text">
    <string>Settings</string>
   </property>
   <property name="toolTip">
    <string>Open settings</string>
   </property>
   <property name="statusTip">
    <string>Open settings</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SceneCanvas</class>
   <extends>QWidget</extends>
   <header>include/scenecanvas.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../icons/icons.qrc"/>
  <include location="../icons/icons.qrc"/>
  <include location="../icons/icons.qrc"/>
 </resources>
 <connections/>
</ui>
